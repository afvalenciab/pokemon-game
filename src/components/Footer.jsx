import React from 'react';
import '../assets/styles/Footer.scss';
import snorlax from '../assets/static/Pokemon_snorlax.svg';

const Footer = () => {
  return (
    <footer className='home--options'>
      <div className='pokeball'>
        <button className='button--play'>Jugar</button>
        <button className='button--restart'>Reiniciar</button>
      </div>
      <figure>
        <img src={snorlax} alt='pokemon snorlax' />
      </figure>
    </footer>
  );
};

export default Footer;
