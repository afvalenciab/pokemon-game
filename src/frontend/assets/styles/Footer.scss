.home--options {
  grid-area: options;
  display: flex;
  justify-content: center;
  align-items: center;

  .pokeball {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    border: 3px solid #000;
    width: 250px;
    height: 250px;
    border-radius: 50%;
    overflow: hidden;
  }

  button {
    border: 0;
    width: 100%;
    height: 50%;
    font-size: 18px;
    font-family: 'Space Mono', monospace;
    box-sizing: border-box;
    outline: none;
    cursor: pointer;
    font-weight: 700;
  }

  .button--play {
    border-bottom: 4px solid #000;
    background-color: #DD0001;
    color: #FFF;
    width: 100%;
    height: 50%;
    transition: .2s;
  }

  .button--play:hover {
    background-color: rgb(165, 4, 4);
  }
  .button--play:active {
    font-size: 14px;
  }

  .button--restart {
    border-top: 4px solid #000;
    width: 100%;
    height: 50%;
    transition: .2s;
  }
  .button--restart:hover {
    background-color: rgba(90,90,90,.2);
  }
  .button--restart:active {
    font-size: 14px;
  }

  figure {
    align-self: flex-end;
    margin: 0;
    img {
      vertical-align: top;
      width: 70px;
    }
  }
}

.home--options.is-active {
  position: relative;
  z-index: 2;
}

@media screen and (max-width: 650px) {
  .home--options {
    .pokeball {
      width: 100px;
      height: 100px;
    }
  
    button {
      font-size: 14px;
    }
  
    .button--play:active {
      font-size: 10px;
    }
  
    .button--restart:active {
      font-size: 10px;
    }
  
    figure {
      img {
        width: 50px;
      }
    }
  }
}