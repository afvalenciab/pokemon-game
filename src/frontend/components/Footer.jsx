import React from 'react';
import snorlax from '../assets/static/Pokemon_snorlax.svg';
import '../assets/styles/Footer.scss';
import counterpart from 'counterpart';
import Translate from 'react-translate-component';
import en from '../lang/en';
import es from '../lang/es';

counterpart.registerTranslations('en', en);
counterpart.registerTranslations('es', es);

const Footer = (props) => {
  const { hanldeClickRestart, statusGame, handleClickPlay, lang } = props;
  counterpart.setLocale(lang ? lang : 'es');

  return (
    <footer className={`home--options ${statusGame==='stop' ? 'is-active': ''}`}>
      <div className='pokeball'>
        <Translate content='play' component='button' className='button--play' onClick={handleClickPlay} />
        <Translate content='restart' component='button'className='button--restart' onClick={hanldeClickRestart} />
      </div>
      <figure>
        <img src={snorlax} alt='pokemon snorlax' />
      </figure>
    </footer>
  );
};

export default Footer;
