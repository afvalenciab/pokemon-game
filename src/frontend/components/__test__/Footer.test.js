import React from 'react';
import { create } from 'react-test-renderer';
import Footer from '../Footer';
import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Footer testing', () => {
  test('Match Snapshot', () => {
    const footer = create(<Footer />);
    expect(footer.toJSON()).toMatchSnapshot();
  });

  test('footer has 2 button tags', () => {
    const footer = render(<Footer />);
    expect(footer.find('button')).toHaveLength(2);
  });

  test('Footer has 1 img tag', () => {
    const footer = render(<Footer />);
    expect(footer.find('img')).toHaveLength(1);
  });
});