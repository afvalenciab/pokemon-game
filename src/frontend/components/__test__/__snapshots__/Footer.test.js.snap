// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer testing Match Snapshot 1`] = `
<footer
  className="home--options "
>
  <div
    className="pokeball"
  >
    <button
      className="button--play"
    >
      Jugar
    </button>
    <button
      className="button--restart"
    >
      Reiniciar
    </button>
  </div>
  <figure>
    <img
      alt="pokemon snorlax"
      src="test-file-stubs"
    />
  </figure>
</footer>
`;
